src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -84,6 +102,20 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
                 {
                     context.ReportDiagnostic(diagnostic);
                 }
+
+                // Due to https://github.com/dotnet/roslyn/issues/41640, enabling this analyzer (IDE0005) on build requires users
+                // to enable generation of XML documentation comments. We detect if generation of XML documentation comments
+                // is disabled for this tree and IDE0005 diagnostics are being reported with effective severity "Warning" or "Error".
+                // If so, we report a special diagnostic that recommends the users to set "GenerateDocumentationFile" to "true"
+                // in their project file to enable IDE0005 on build.
+                if (tree.Options.DocumentationMode == DocumentationMode.Parse)
```

Comment:
💭 The other option here would be checking for `DocumentationMode.None`. This way, we only report the diagnostic if the user has enabled code style on build _and_ set IDE0005 to warning _and_ hasn't enabled documentation output.

If we check for `DocumentationMode.Parse`, it seems like we should also to make sure code style is enabled on build.

Reply:
That sounds a better approach! I have adjusted the implementation accordingly.